cmd/kubelet/app/server.go

```diff
@@ -488,13 +493,19 @@ func (s *KubeletServer) kubeconfigClientConfig() (*client.Config, error) {
 // fall back to the default auth (none) without an error.
 // TODO(roberthbailey): Remove support for --auth-path
 func (s *KubeletServer) createClientConfig() (*client.Config, error) {
-	if s.KubeConfig.Provided() && s.AuthPath.Provided() {
-		return nil, fmt.Errorf("cannot specify both --kubeconfig and --auth-path")
-	}
-	if s.KubeConfig.Provided() {
-		return s.kubeconfigClientConfig()
-	} else if s.AuthPath.Provided() {
-		return s.authPathClientConfig(false)
+	authPathFlag := pflag.Lookup(authPathFlagName)
+	kubeconfigFlag := pflag.Lookup(kubeconfigFlagName)
+	if authPathFlag != nil && kubeconfigFlag != nil {
+		if authPathFlag.Changed && kubeconfigFlag.Changed {
```

Comment:
Because the operation of the command depends on being able to distinguish between set-but-empty (to clear a value) and unset-and-empty (noop). An explicit param to clear a value would allow an all-primitives options struct with no external command line flag lookups